src/main/steps/c100-rebuild/miam/upload/postController.ts

```diff
@@ -0,0 +1,104 @@
+import autobind from 'autobind-decorator';
+import { Response } from 'express';
+import FormData from 'form-data';
+import { isNull } from 'lodash';
+
+import { DocumentUploadResponse } from '../../../../app/case/C100CaseApi';
+import { C100DocumentInfo } from '../../../../app/case/definition';
+import { AppRequest } from '../../../../app/controller/AppRequest';
+import { AnyObject, PostController } from '../../../../app/controller/PostController';
+import { FormFields, FormFieldsFn } from '../../../../app/form/Form';
+import { C100_MIAM_UPLOAD } from '../../../urls';
+
+import { AnyType } from './getController';
+
+// eslint-disable-next-line import/no-unresolved
+//import { AnyType } from '../getController';
+
+/* The UploadDocumentController class extends the PostController class and overrides the
+PostDocumentUploader method */
+@autobind
+export default class UploadDocumentController extends PostController<AnyObject> {
+  constructor(protected readonly fields: FormFields | FormFieldsFn) {
+    super(fields);
+  }
+
+  /**
+   * A recursive function that calls itself.
+   * @param req - AppRequest<AnyObject>
+   * @param {Response} res - Response - This is the response object that will be sent back to the client.
+   */
+  public async post(req: AppRequest<AnyObject>, res: Response): Promise<void> {
+    const { files }: AppRequest<AnyObject> = req;
+    const miamCertificate = req.session.userCase?.miam_certificate as C100DocumentInfo;
+
+    if (this.checkIfDocumentAlreadyExist(miamCertificate)) {
+      req.session.errors = [{ propertyName: 'document', errorType: 'multipleFiles' }];
+      req.session.save(err => {
```

Comment:
@SangeethaPonnusamy We need to do other validations for file format & max file size as done in [PRLC100-202](https://tools.hmcts.net/jira/browse/PRLC100-202) by @kumarmadurai

Reply:
fixed it